tower-grpc-examples/src/metadata/server.rs

```diff
@@ -0,0 +1,74 @@
+extern crate bytes;
+extern crate env_logger;
+extern crate futures;
+#[macro_use]
+extern crate log;
+extern crate prost;
+#[macro_use]
+extern crate prost_derive;
+extern crate tokio_core;
+extern crate tower_h2;
+extern crate tower_grpc;
+extern crate http;
+
+pub mod metadata {
+    include!(concat!(env!("OUT_DIR"), "/metadata.rs"));
+}
+
+use metadata::{server, EnterRequest, EnterReply};
+
+use futures::{future, Future, Stream};
+use tokio_core::net::TcpListener;
+use tokio_core::reactor::Core;
+use tower_h2::Server;
+use tower_grpc::{Request, Response};
+
+#[derive(Clone, Debug)]
+struct Door;
+
+impl server::Doorman for Door {
+    type AskToEnterFuture = future::FutureResult<Response<EnterReply>, tower_grpc::Error>;
+
+    fn ask_to_enter(&mut self, request: Request<EnterRequest>) -> Self::AskToEnterFuture {
+        println!("REQUEST = {:?}", request);
+
+        let message = match request.headers().get("metadata").and_then(|header| header.to_str().ok()) {
```

Comment:
Nit: would be nice if this was wrapped, or broken into something like:
```rust
let metadata = request.headers()
    .get("metadata")
    .and_then(|header| header.to_str().ok());
let message = match metadata {
    // ...
};
```

Reply:
haha, I had that at first and then change it to the one liner...